cmds/core/wget/wget_test.go

```diff
@@ -98,25 +99,20 @@ var tests = []struct {
 	},
 }
 
-func getFreePort(t *testing.T) int {
+func getFreePort(t *testing.T) (net.Listener, int) {
 	l, err := net.Listen("tcp", ":0")
 	if err != nil {
 		t.Fatalf("Cannot create free port: %v", err)
 	}
-	l.Close()
-	return l.Addr().(*net.TCPAddr).Port
+	return l, l.Addr().(*net.TCPAddr).Port
 }
 
 // TestWget implements a table-driven test.
 func TestWget(t *testing.T) {
 	// Start a webserver on a free port.
-	unusedPort := getFreePort(t)
-
-	l, err := net.Listen("tcp", ":0")
-	if err != nil {
-		t.Fatalf("Cannot create free port: %v", err)
-	}
-	port := l.Addr().(*net.TCPAddr).Port
+	l, port := getFreePort(t)
+	ul, unusedPort := getFreePort(t)
+	ul.Close()
```

Comment:
Won't this result in the same problem as described in the PR? Something else might come along and start listening on this "unusedPort" and cause the test to fail. Is there a better/different way to force the failure?

Reply:
I tired to call ul.Close() at the end but the test will hang. Maybe after opening the port, we just don't close it. WDYT?

Reply:
I'm thinking about not testing on closed port in the test case. We just test a hostname that doesn't exist. This can prevent the hang issue also the used port issue.

Reply:
hehe, I forgot the "defer listener.Close()" statement for the bad server in my example code :)

Reply:
Thanks for the suggestions. Creating a fake server that closes the connection is a good idea. However, I think the original testing method is still good since it can verify different IP format so I keep it.